fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java

```diff
@@ -475,6 +479,39 @@ public List<Long> createShards(int numShards, FilePathInfo pathInfo, FileCacheIn
         return shardInfos.stream().map(ShardInfo::getShardId).collect(Collectors.toList());
     }
 
+    public void updateShardGroup(LakeTable table, boolean enableCache) throws DdlException {
+        Collection<Partition> partitions = table.getPartitions();
+        List<Long> groupIds =
+                partitions.stream().mapToLong(partition -> partition.getShardGroupId()).boxed().collect(Collectors.toList());
+        try {
+            List<UpdateShardGroupInfo> updateShardGroupInfoList = new ArrayList<>(groupIds.size());
+            for (Long groupId : groupIds) {
+                updateShardGroupInfoList.add(
+                        UpdateShardGroupInfo.newBuilder()
+                                .setGroupId(groupId)
+                                .setEnableCache(enableCache ? CacheEnableState.ENABLED : CacheEnableState.DISABLED)
+                                .build());
+            }
+
+            client.updateShardGroup(serviceId, updateShardGroupInfoList);
```

Comment:
consider what to do with the partial success case.

Reply:
If any exception occurs during the updateShardGroup process, all shards will be rolled back to their original state, so there is no partial sucess.